Tests/Logic/Search/RKSearchTest.m

```diff
@@ -60,4 +60,51 @@ - (void)testSearchingForManagedObjects
+{
+    __block NSError *error;
+    NSURL *modelURL = [[RKTestFixture fixtureBundle] URLForResource:@"Data Model" withExtension:@"mom"];
+    NSManagedObjectModel *managedObjectModel = [[NSManagedObjectModel alloc] initWithContentsOfURL:modelURL];
+    RKManagedObjectStore *managedObjectStore = [[RKManagedObjectStore alloc] initWithManagedObjectModel:managedObjectModel];
+    [managedObjectStore addSearchIndexingToEntityForName:@"Cat" onAttributes:@[ @"name" ]];
+    [managedObjectStore addInMemoryPersistentStore:&error];
+    [managedObjectStore createManagedObjectContexts];
+    [managedObjectStore startIndexingPersistentStoreManagedObjectContext];
+    
+    // Get some content into the index
+    RKCat *cat1 = [NSEntityDescription insertNewObjectForEntityForName:@"Cat" inManagedObjectContext:managedObjectStore.mainQueueManagedObjectContext];
+    cat1.name = @"$";
+    RKCat *cat2 = [NSEntityDescription insertNewObjectForEntityForName:@"Cat" inManagedObjectContext:managedObjectStore.mainQueueManagedObjectContext];
+    cat2.name = @"£ sterling";
+    RKCat *cat3 = [NSEntityDescription insertNewObjectForEntityForName:@"Cat" inManagedObjectContext:managedObjectStore.mainQueueManagedObjectContext];
+    cat3.name = @"€ euro";
+    
+    [managedObjectStore.mainQueueManagedObjectContext obtainPermanentIDsForObjects:@[cat1, cat2] error:&error];
+    [managedObjectStore.mainQueueManagedObjectContext saveToPersistentStore:&error];
+    
+    // Execute fetches to verify
+    [managedObjectStore.persistentStoreManagedObjectContext performBlockAndWait:^{
+        NSPredicate *predicate = [RKSearchPredicate searchPredicateWithText:@"$" type:NSAndPredicateType];
+        
+        NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Cat"];
+        fetchRequest.predicate = predicate;
+        NSArray *objects = [managedObjectStore.persistentStoreManagedObjectContext executeFetchRequest:fetchRequest error:&error];
+        assertThat(objects, hasCountOf(1));
+        assertThat([objects[0] objectID], is(equalTo(cat1.objectID)));
+    }];
+    
+    [managedObjectStore.persistentStoreManagedObjectContext performBlockAndWait:^{
+        NSPredicate *predicate = [RKSearchPredicate searchPredicateWithText:@"$ £ €" type:NSOrPredicateType];
+        
+        NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:@"Cat"];
+        fetchRequest.predicate = predicate;
+        fetchRequest.sortDescriptors = @[ [NSSortDescriptor sortDescriptorWithKey:@"name" ascending:YES] ];
+        NSArray *objects = [managedObjectStore.persistentStoreManagedObjectContext executeFetchRequest:fetchRequest error:&error];
+        assertThat(objects, hasCountOf(3));
+        assertThat([objects[0] objectID], is(equalTo(cat1.objectID))); // $
+        assertThat([objects[1] objectID], is(equalTo(cat2.objectID))); // £
+        assertThat([objects[1] objectID], is(equalTo(cat2.objectID))); // €
```

Comment:
did you mean `assertThat([objects[2] objectID], is(equalTo(cat3.objectID)));` ?

Reply:
I certainly did. 💻👀 Fixed and pull request updated.